website/content/docs/concepts/secure-variables.mdx

```diff
@@ -0,0 +1,124 @@
+---
+layout: docs
+page_title: Secure Variables
+description: Learn about Nomad's secure variables feature
+---
+
+# Secure Variables
+
+Most Nomad workloads need configuration or secrets. Nomad provides a
+`template` block to run consul-template alongside tasks, but leaves the
+source of those configuration values up to the operator. Typically
+you'll use [HashiCorp Vault] as the source for secrets.
+
+Nomad's Secure Variables feature provides the option to store secrets
+at file-like paths directly in Nomad's state store. The contents of
+secrets are encrypted and replicated between servers with raft. Access
+to secrets is controlled by ACL policies, and tasks have implicit ACL
+policies that allow them to access their own secrets. You can create,
+read, update, or delete secrets via the command line or in the Nomad
+web UI.
+
+Note that the Secure Variables feature is intended for small pieces of
+configuration data needed by workloads. Because writing to the Nomad
+state store uses resources needed by Nomad, it's not well-suited for
+large or fast-changing data like batch job results. These should be
+stored in a separate database. Secure Variables are also not intended
+to be a full replacement for HashiCorp Vault. If you need powerful
+options like dynamic secrets or transit encryption, you'll need to use
+Vault.
+
+## ACL for Secure Variables
+
+Every Secure Variable belongs to a specific Nomad namespace. ACL
+policies can restrict access to Secure Variables within a namespace on
+a per-path basis, using a list of `path` blocks, under the
+`namespace.secure_variables` block.
+
+Path definitions may also include wildcard symbols, also called globs,
+allowing a single path policy definition to apply to a set of paths
+within that namespace. For example, the policy below allows full
+access to secure variables at all paths in the "dev" namespace that
+are prefixed with "project/", but only read access to paths prefixed
+with "system/":
+
+```hcl
+namespace "dev" {
+  policy       = "write"
+  capabilities = ["alloc-node-exec"]
+
+  secure_variables {
+
+    # full access to secrets in all "project" paths
+    path "project/*" {
```

Comment:
`question`: Should we mention anything about case-sensitivity, whitespace and acceptable characters for the `path`?

Reply:
That's a good question... right now details like this and your question about prefix matching below _should_ be the [Learn Guide on Policies](https://learn.hashicorp.com/tutorials/nomad/access-control-policies?in=nomad/access-control) but aren't! 

Unfortunately it doesn't look like we document the schema at that level of detail anywhere 😦 . I've been musing working up a ACL Policy Reference doc that takes the Learn Guide stuff and makes it into a detailed reference for the whole ACL policy schema. I'll do that in a separate PR, because we'll want to backport that so it shows for all versions of Nomad. Then we can refer to that new document from here to pick up all these good details.

Reply:
That reference doc is https://github.com/hashicorp/nomad/pull/13787 and I'll update this PR with links once that's merged.